pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -292,15 +304,33 @@ func constructDatastoreContext(config types.DownloaderConfig, status *types.Down
 			downloadURL = downloadURL + "/" + config.Name
 		}
 	}
+
+	cred := types.CredentialBlock{
+		Identity: dst.ApiKey,
+		Password: dst.Password,
+	}
+	if dst.IsCipher && len(dst.CipherBlock.CipherData) != 0 {
+		plainText, err := tpmmgr.DecryptWithCipherInfo(dst.CipherBlock)
+		if len(plainText) != 0 && err == nil {
+			err = json.Unmarshal([]byte(plainText), &cred)
+			if err != nil {
+				errStr := fmt.Sprintf("%v", err)
+				return nil, errStr
```

Comment:
Errors are not passed as strings in go. They are passed as the type error.
You can prepend information to an existing error if need be, but here you do not prepend anything.
So please just define this function to return error and not string.